caffe2/python/operator_test/mkl_alexnet_test.py

```diff
@@ -0,0 +1,337 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+import unittest
+
+import numpy as np
+from caffe2.proto import caffe2_pb2
+from caffe2.python import cnn, core, workspace, test_util
+
+
+@unittest.skipIf(not workspace.C.has_mkldnn, "Skipping as we do not have mkldnn.")
+class TestAlexNet(test_util.TestCase):
+    bs = 32
+    layerwisetiming = False
+
+    def testMKLAlexNet(self):
+        bs = self.bs
+        layerwisetiming = self.layerwisetiming
+        
+        X = np.random.rand(bs, 3, 224, 224).astype(np.float32) - 0.5
+        W1 = np.random.rand(64, 3, 11, 11).astype(np.float32) - 0.5
+        b1 = np.random.rand(64).astype(np.float32) - 0.5
+        
+        W2 = np.random.rand(192, 64, 5, 5).astype(np.float32) - 0.5
+        b2 = np.random.rand(192).astype(np.float32) - 0.5
+
+        W3 = np.random.rand(384, 192, 3, 3).astype(np.float32) - 0.5
+        b3 = np.random.rand(384).astype(np.float32) - 0.5
+
+        W4 = np.random.rand(256, 384, 3, 3).astype(np.float32) - 0.5
+        b4 = np.random.rand(256).astype(np.float32) - 0.5
+
+        W5 = np.random.rand(256, 256, 3, 3).astype(np.float32) - 0.5
+        b5 = np.random.rand(256).astype(np.float32) - 0.5
+
+        Wfc1 = np.random.rand(4096, 256*6*6).astype(np.float32) - 0.5
+        bfc1 = np.random.rand(4096).astype(np.float32) - 0.5
+        
+        Wfc2 = np.random.rand(4096, 4096).astype(np.float32) - 0.5
+        bfc2 = np.random.rand(4096).astype(np.float32) - 0.5
+
+        Wfc3 = np.random.rand(1000, 4096).astype(np.float32) - 0.5
+        bfc3 = np.random.rand(1000).astype(np.float32) - 0.5
+
+        label = np.random.randint(low=0, high=1000, size=(bs,)).astype(np.int32)
+
+        mkl_do = core.DeviceOption(caffe2_pb2.MKLDNN)
+     
+
+        #feed mkl
+        workspace.FeedBlob("Xmkl", X, device_option=mkl_do)
```

Comment:
did you consider using CNNModelHelper instead of manually creating all the parameters? Similar to convnet_benchmarks.py